bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java

```diff
@@ -28,55 +36,81 @@
  * @author Kai Kreuzer - Initial contribution and API
  *
  */
-public class SitemapProviderImpl implements SitemapProvider {
+@Component(service = SitemapProvider.class)
+public class SitemapProviderImpl implements SitemapProvider, ModelRepositoryChangeListener {
 
-    protected static final String SITEMAP_FILEEXT = ".sitemap";
+    private static final String SITEMAP_MODEL_NAME = "sitemap";
+    protected static final String SITEMAP_FILEEXT = "." + SITEMAP_MODEL_NAME;
 
     private final Logger logger = LoggerFactory.getLogger(SitemapProviderImpl.class);
 
     private ModelRepository modelRepo = null;
 
+    private final Map<String, Sitemap> sitemapModelCache = new ConcurrentHashMap<>();
+
+    @Reference
     public void setModelRepository(ModelRepository modelRepo) {
         this.modelRepo = modelRepo;
+        refreshSitemapModels();

```

Comment:
Here again, it would be much nicer to do this in activate() when the component is really meant to become active.